flask_oauthlib/provider/oauth2.py

```diff
@@ -622,9 +622,12 @@ def authenticate_client(self, request, *args, **kwargs):
 
         request.client = client
 
-        if client.client_secret != client_secret:
-            log.debug('Authenticate client failed, secret not match.')
-            return False
+        # http://tools.ietf.org/html/rfc6749#section-2
+        # The client MAY omit the parameter if the client secret is an empty string.
+        if hasattr(client, 'client_secret'):
+            if client.client_secret != client_secret:
```

Comment:
Why not `if hasattr(client, 'client_secret') and client.client_secret != client_secret`?